### PR TITLE
Action log percept

### PR DESCRIPTION
**Pull request by StefanV7**
Added actionlog percept and some tests.

User story:
As a programmer I want to be able to use the actionlog percept so that I can obtain information on the influence of new buildings on my indicators.

### CODE REVIEW

**Comment by dionytadema** [comment]
i still agree 👍

File: Municipality/tygronEvents.mod2g [hunk 1]
```diff
@@ -11,6 +11,17 @@ module tygronEvents {
 	forall percept(indicators(List)), bel(indicator(ID,OldCurrent,OldTarget), member(indicator(ID,NewCurrent,NewTarget),List)) 
 		do delete(indicator(ID,OldCurrent,OldTarget)) + insert(indicator(ID,NewCurrent,NewTarget)).
 	
+	%%% ACTIONLOG PERCEPT
+	%%% Get information about the change in indicators of performed actions.
+	
+	%%% Delete the old one for each stakeholder
+	forall percept(action_logs(ActList)), bel(member(actionlog(StakeholderID,_,_,_),ActList), actionlog(StakeholderID,A,B,C))
+		do delete(actionlog(StakeholderID,A,B,C)).
```
Comment (NickTUD) [inline Municipality/tygronEvents.mod2g]: :-1: Update variable names to more descriptive ones (for only the code you changed). This only accounts for variable names with lengths <= 2.

File: Municipality/test.test2g [hunk 2]
```diff
@@ -22,13 +22,19 @@ test tygron with
 		percept(stakeholders(List)), bel(member(SubList, List), 
 			member(indicatorLink(ShID,IndWeights), SubList)))) 
 			leadsto bel(indicatorLink(ShID,IndWeights)).
+		percept(action_logs(ActList)), bel(member(actionlog(StakeholderID,Description,ActionID,IncList),ActList))
+			leadsto bel(actionlog(StakeholderID,Description,ActionID,IncList)).
+		percept(functions(FList)), bel(member([Name,Id,L], FList))
```
Comment (NickTUD) [inline Municipality/test.test2g]: Here as well!

File: Municipality/tygronEvents.mod2g [hunk 3]
```diff
@@ -11,6 +11,17 @@ module tygronEvents {
 	forall percept(indicators(List)), bel(indicator(ID,OldCurrent,OldTarget), member(indicator(ID,NewCurrent,NewTarget),List))
```
Comment (NickTUD) [inline Municipality/tygronEvents.mod2g]: ID to IndicatorID

File: Municipality/test.test2g [hunk 4]
```diff
@@ -22,13 +22,19 @@ test tygron with
...
+			leadsto bel(function(Name,Id,L)).
 		
 		%If a percept list is perceived, which doesn't contain a percept which we first belief was true, delete that belief.
 		%These tests have been made according to the templates in the GOAL Programming guide.
 		percept(buildings(List)), bel(building(ID,Name,OwnerID,ConstructionYear,CategoriesList,Floors),
 			not(member(building(ID,Name,OwnerID,ConstructionYear,CategoriesList,Floors),List)))
```
Comment (NickTUD) [inline Municipality/test.test2g]: ID to BuildingID

**Comment by NickTUD** [comment]
If the build succeeds :+1:

**Comment by StefanV7** [comment]
Preferably you could have waited for the thumbs up from Nick, but it's ok I guess.

**Pull request merged**